File: LeetCode/Python/Medium/1249-Minimum-Remove-to-Make-Valid-Parentheses/solution.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        result=""

        s_characters=[character for character in s]

        stack=list()

        for (index, character) in enumerate(s_characters):
            if character=='(':
                stack.append(index)
            elif character==')':
                if stack:
                    stack.pop()
                else:
                    s_characters[index]='$'

        while stack:
            index=stack.pop()

            s_characters[index]='$'

        result_characters=list()

        for character in s_characters:
            if character!='$':
                result_characters.append(character)

        result="".join(result_characters)

        return result

File: LeetCode/Rust/Medium/1249-Minimum-Remove-to-Make-Valid-Parentheses/solution.rs
impl Solution
{
    pub fn min_remove_to_make_valid(s: String) -> String
    {
        let mut result: String=String::from("");

        let mut s_characters: Vec<char>=s.chars().collect::<Vec<char>>();

        let mut stack: Vec<usize>=Vec::new();

        for (index, character) in s.char_indices()
        {
            if character=='('
            {
                stack.push(index);
            }
            else if character==')'
            {
                if !stack.is_empty()
                {
                    stack.pop();
                }
                else
                {
                    s_characters[index]='$';
                }
            }
        }

        while let Some(index)=stack.pop()
        {
            s_characters[index]='$';
        }

        result=s_characters.into_iter().filter(|character| *character!='$').collect::<String>();

        result
    }
}